shared_model/backend/protobuf/queries/proto_ordering.hpp

```diff
@@ -0,0 +1,49 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef IROHA_SHARED_PROTO_MODEL_QUERY_ORDERING_HPP
+#define IROHA_SHARED_PROTO_MODEL_QUERY_ORDERING_HPP
+
+#include "interfaces/queries/ordering.hpp"
+#include "interfaces/common_objects/types.hpp"
+
+#include "queries.pb.h"
+
+namespace shared_model {
+  namespace proto {
+
+    /// Provides ordering data for pagination
+    class OrderingImpl final : public interface::Ordering {
+    public:
+      using BaseType = interface::Ordering;
+
+      OrderingImpl& operator=(OrderingImpl const&) = delete;
+      OrderingImpl& operator=(OrderingImpl&&) = delete;
+
+      OrderingImpl();
+      OrderingImpl(iroha::protocol::Ordering* proto_ordering);
+      OrderingImpl(OrderingImpl&&);
+      OrderingImpl(OrderingImpl const&);
+
+      bool insert(BaseType::Field field, BaseType::Direction direction) override;
+      void reset() override;
+      void get(BaseType::OrderingEntry const*& orderingEntry, size_t& count) const override;
+
+    private:
+      size_t count_;
+      bool inserted_[(size_t)BaseType::Field::kMaxValueCount];
+      BaseType::OrderingEntry ordering_[(size_t)BaseType::Field::kMaxValueCount];
+
+      static_assert(sizeof(inserted_)/sizeof(inserted_[0]) == sizeof(ordering_)/sizeof(ordering_[0]), 
+        "inserted_ and ordering_ must be the same size.");
```

Comment:
i'd move this assertion to the place where it is required - probably to `OrderingImpl::insert_(...)`
this will help others understand why it is asserted and clear the header a bit

Reply:
This is not run-time check, but compile-time. To prevent the posibility of independent changes.